.github/workflows/automate-changelog-PR.yml

```diff
@@ -0,0 +1,48 @@
+name: "Update changelog and open a pull request"
+
+on:
+  workflow_dispatch:
+    inputs:
+      fname_changelog:
+        description: 'Changelog filename'
+        required: true
+        default: 'CHANGES.md'
+      milestone_title:
+        description: 'Milestone title'
+        required: true
+
+jobs:
+  create-changelog-PR:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Python (to install 'changelog')
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Checkout 'changelog'
+      uses: actions/checkout@v2
+      with:
+        repository: neuropoly/changelog
+        path: changelog
+    - name: Install 'changelog'
+      run: |
+        pip install ./changelog
+    - name: Checkout '${{ github.event.repository.name }}'
+      uses: actions/checkout@v2
+      with:
+        path: ${{ github.event.repository.name }}
+    - name: Update '${{ github.event.inputs.fname_changelog }}' for '${{ github.event.repository.name }}'
+      run: |
+        cd ${{ github.event.repository.name }}
+        changelog neuropoly/spinalcordtoolbox --update --name "${{ github.event.inputs.fname_changelog }}" --milestone ${{ github.event.inputs.milestone_title }}
+        rm "${{ github.event.inputs.fname_changelog }}.bak"
+    - name: Create pull request for updated changelog
+      uses: peter-evans/create-pull-request@v3
+      with:
+        path: "${{ github.event.repository.name }}"
+        branch: "bot/v${{ github.event.inputs.milestone_title }}"
+        commit-message  : "Update ${{ github.event.inputs.fname_changelog }} for ${{ github.event.inputs.milestone_title }} release"
+        title: "Update ${{ github.event.inputs.fname_changelog }} for ${{ github.event.inputs.milestone_title }} release"
+        milestone: "${{ github.event.inputs.milestone_title }}"
+        body: "Automated step carried out as a part of [Creating a new release](https://github.com/neuropoly/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release) checklist."
```

Comment:
Do you this it would be possible to *tag the Project*, so that CHANGES.md shows up in e.g. https://github.com/neuropoly/spinalcordtoolbox/projects/15 ?

Reply:
I wish this were possible. I think the core problem here is that GitHub has poor integration between milestones and project boards.

What I _really_ wish I could do is link a project board to a milestone. I want to be able to have any issues/PRs tagged with a milestone to be automatically added to the corresponding project board.

(Without this, we're running into friction where Julien will add a milestone to an issue/PR, then wonder why it's not also on the project board... we're kind of using 2 different methods for release tracking...)

Reply:
Ohhh... wait just one minute. https://github.com/marketplace/actions/move-labeled-or-milestoned-issue-to-a-specific-project-column

Reply:
I think I could create a GH actions workflow that triggers when an issue/PR is milestoned, then adds it to a project board of the same name. (Which would solve both issues described above.)

Reply:
Opened https://github.com/neuropoly/spinalcordtoolbox/issues/3338. Since this is a larger issue with our use of project boards, I'm going to mark this as "resolved" for now.